### ISSUE TITLE
TreeBrowser isn't easily extensible

### ISSUE THREAD

**Issue by buckett** [instructure/instructure-ui#1177]
### Context
TreeBrowser

### Summary
The TreeBrowser component works nicely however I got to the point of wanting to add a dropdown `<Menu>` to the items to allow actions on the items in the tree, however there doesn't seem to be an easy way to add this.

I can supply a `renderContent` property however then I end up re-implementing the rendering of icons (item/collections, the expanded/close) and the text layout (name/description), it would be nice if there was a way of adding to the existing behaviour.

### Justification
Displaying a tree is useful, but then it's also useful to have actions for items/collections in the tree.

### Additional Details

I'll probably just use the selection to handle this and then put the UI elements outside the TreeBrowser (in a similar way to how the Files tool in Canvas handles things like View/Manage Access/Download/etc) but this isn't how I'd like it to work.

**Comment by ke1k0** [comment]
I've talked to the team, and we decided it is a low priority addition to the component at the moment. 

TreeBrowser is technically not a deprecated component, but one of the reasons why the Design team created the [Drilldown](https://instructure.design/#Drilldown) component was to replace TreeBrowser eventually. Drilldown is responsive, while using TreeBrowser on smaller viewports can be challenging. Drilldown is also way better at handling actions.

I don't know of any short-term plans to replace all existing TreeBrowsers with the Drilldown pattern, but we are not planning to further improve on the TreeBrowser either, unless it is necessary, or the Design team decides to focus on this component again. 

If there is something specific we can help with, let us know :)
